### PR TITLE
Add Mkdocs API site

### PR DESCRIPTION
**Pull request by pauloue**
A friendlier display of `lua_api.txt`. See #7084.

I used GitLab Pages because it doesn't require using a gh-pages branch or giving Travis access to the repo.

Tested here: https://pauloue.gitlab.io/minetest/.

### CODE REVIEW

File: mkdocs/build.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+PYGMENTS_FILE=$(pip show pygments | awk '/Location/ { print $2 }')/pygments/formatters/html.py
+MARKDOWN_FILE=$(pip show markdown | awk '/Location/ { print $2 }')/markdown/extensions/codehilite.py
+patch -N -r - $PYGMENTS_FILE code_tag.patch
+patch -N -r - $MARKDOWN_FILE lua_highlight.patch
+
+# Split lua_api.txt on top level headings
+cat ../doc/lua_api.txt | csplit -sz -f docs/section -b '%02d.md' - '/^=/-1' '{*}'
+
+cat > mkdocs.yml << EOF
+site_name: Mintest API Documentation
```
Comment (ClobberXD) [inline mkdocs/build.sh]:
```suggestion
site_name: Minetest API Documentation
```

:)

**Comment by ClobberXD** [comment]
The folder trees look a little weird. See the mod directory structure section for an example: https://pauloue.gitlab.io/minetest/mods/#mod-directory-structure

**Comment by pauloue** [comment]
I already fixed that (see the force-push), it just didn't update due to a build failure.

**Comment by SmallJoker** [comment]
The implementation looks good (demo site). Could you please use tabs for the shell script, like all other util scripts?
I'm not sure about the other files, but tabs should be used where's possible.

**Comment by paramat** [comment]
This is a very good idea.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -322,3 +322,18 @@ package:docker:
     - docker push ${CONTAINER_IMAGE}/server:$CI_COMMIT_SHA
     - docker push ${CONTAINER_IMAGE}/server:$CI_COMMIT_REF_NAME
     - docker push ${CONTAINER_IMAGE}/server:latest
+
+pages:
+  stage: deploy
+  image: python:3.6
```
Comment (nerzhul) [inline .gitlab-ci.yml]: python3.7 please

**Comment by pauloue** [comment]
The CI stopped working for some reason: https://gitlab.com/pauloue/minetest/-/jobs/163549298 (not because of python3.7). It worked 3 weeks ago: https://gitlab.com/pauloue/minetest/-/jobs/151355185.

**Comment by pauloue** [comment]
There was something wrong with the latest version of pip.

**Comment by nerzhul** [comment]
See http://minetest.gitlab.io/minetest/ (Gitlab autogenerated URL)

**Pull request merged**